### ISSUE TITLE
ModuleNotFoundError: No module named 'lldb.embedded_interpreter'

### ISSUE THREAD

**Issue by yangyibiao** [llvm/llvm-project#55575]
lldb is already the newest version (1:15.0-55~20220216125911.2).

```console
$ lldb --version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'lldb.embedded_interpreter'
lldb version 15.0.0
```

**Comment by JDevlieghere** [comment]
We'll need a little bit more information. What platform are you on? How did you install lldb?

**Comment by yangyibiao** [comment]
> We'll need a little bit more information. What platform are you on? How did you install lldb?

**Comment by nlsandler** [comment]
I see the same behavior in LLDB 14 on Ubuntu 20.04 (not running in a container), installed with
`bash -c "$(wget -O - https://apt.llvm.org/llvm.sh)"`

**Comment by michael-swan** [comment]
I suspect OP's issue is the same as mine. `lldb -P` returns `/usr/lib/lib/python3.8/site-packages` for LLDB builds from `deb http://apt.llvm.org/focal/ llvm-toolchain-focal-14` repos. It should instead return `/usr/lib/llvm-14/lib/python3.8/site-packages` which is where these Python dependencies are installed. Any ideas where this mistake originates?

**Comment by fonqL** [comment]
Same here. Win10 wsl2. After updating Ubuntu from 20.04 to 22.04, I found lldb still kept version 10 so I ran `apt-get -u dist-upgrade` and upgraded again. lldb turned to version 14.0.0 now but came with this problem.

**Comment by amordo** [comment]
Thanks for the link, @tgfrerer 
tldr: `lldb -P` prints destination, which doesn't exist; do `mkdir -p /usr/lib/local/lib/python3.10 && ln -s /usr/lib/llvm-14/lib/python3.10/dist-packages /usr/lib/local/lib/python3.10/dist-packages`
maybe your `llvm-14` dir is located/named different🙂

**Comment by jason-kane** [comment]
with ubuntu 22.04 and swift 5.7 I had to:

    apt install python3-lldb-14
    ln -s /usr/lib/llvm-14/lib/python3.10/dist-packages/lldb/* /usr/lib/python3/dist-packages/lldb/

before "swift repl" would work without errors.

**Comment by junaire** [comment]
> with ubuntu 22.04 and swift 5.7 I had to:
> 
> ```
> apt install python3-lldb-14
> ln -s /usr/lib/llvm-14/lib/python3.10/dist-packages/lldb/* /usr/lib/python3/dist-packages/lldb/
> ```
> 
> before "swift repl" would work without errors.

I'm on ubuntu22, installed lldb by apt. this works for me. Thanks!

**Comment by unlsycn** [comment]
> with ubuntu 22.04 and swift 5.7 I had to:
> 
> ```
> apt install python3-lldb-14
> ln -s /usr/lib/llvm-14/lib/python3.10/dist-packages/lldb/* /usr/lib/python3/dist-packages/lldb/
> ```
> 
> before "swift repl" would work without errors.

works for me on WSL2. thx!

**Comment by miguno** [comment]
Same here on WSL2 with Ubuntu 22.04:

```shell
 $ lldb --version
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'lldb.embedded_interpreter'
lldb version 14.0.0
```
```shell
 $ lldb -P
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'lldb.embedded_interpreter'
/usr/lib/local/lib/python3.10/dist-packages
 $ ls -l /usr/lib/local/lib/python3.10/dist-packages
ls: cannot access '/usr/lib/local/lib/python3.10/dist-packages': No such file or directory
```

**Comment by albertlarsan68** [comment]
If you aren't admin, you can work around this by setting your `PYTHONPATH` env variable to `/usr/lib/llvm-14/lib/python3.10/dist-packages`.

**Comment by castaway-** [comment]
Same here with Ubuntu v22.10 / lldb v15.0.6 ;)

lldb -P would give me:
`Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'lldb.embedded_interpreter'
/usr/lib/local/lib/python3.10/dist-packages
`

Workaround using proper path for lldb v15
`sudo ln -s /usr/lib/llvm-15/lib/python3.10/dist-packages/lldb /usr/lib/local/lib/python3.10/dist-packages/lldb`

the correct path should use "llvm-xx" instead of "local", that was the only diff
Thanks for the temporary fix !

**Comment by Selyss** [comment]
> ```
> ln -s /usr/lib/llvm-14/lib/python3.10/dist-packages/lldb/* /usr/lib/python3/dist-packages/lldb/
> ```
worked for me on WSL2, thanks!

**Comment by fwqaaq** [comment]
```bash
sudo ln -s /usr/lib/llvm-14/lib/python3.10/dist-packages/lldb/* /usr/lib/python3/dist-packages/lldb/
```

Thanks, and add `sudo` for this command.

**Comment by loop-evgeny** [comment]
The workaround works, but it would be great to fix this finally. This has been open for over a year and has wasted time for a lot of people.

**Comment by JinjieZhao** [comment]
export PYTHONPATH=/lib/llvm-14/lib/python3.10/dist-packages:$PYTHONPATH
it works.

**Comment by createyourpersonalaccount** [comment]
There's a bug report on Debian 12 on this here <https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1050049>. You can append

```sh
export PYTHONPATH="/usr/lib/llvm-14/lib/python3.11/dist-packages/:$(PYTHONPATH)"
```

to your `~/.bashrc`.

**Comment by loop-evgeny** [comment]
Even with the workaround, this is quite annoying when trying to automate LLDB installation, because the LLDB version and Python versions differ between Ubuntu versions, so you have to script detecting the installed versions somehow. 

Is this something LLDB developers can fix or is it a packaging issue?

**Comment by GitMensch** [comment]
Both...

You can commonly work around this by switching to the LLDB version that does not have that module any more

```sh
sudo apt install lldb-16
sudo update-alternatives --install /usr/bin/lldb lldb /usr/bin/lldb-16 50
```

fixed that for me
